src/IIIFPresentation/DLCS/API/DlcsApiClient.cs

```diff
@@ -84,13 +91,63 @@ public async Task<List<Batch>> IngestAssets<T>(int customerId, List<T> assets, C
         return batches.ToList();
     }
 
-    private async Task<T?> CallDlcsApi<T>(HttpMethod httpMethod, string path, object payload,
+    public async Task<List<JObject>> GetBatchAssets(int customerId, int batchId,
+        CancellationToken cancellationToken = default)
+    {
+        logger.LogTrace("Requesting assets for batch {BatchId} for customer {CustomerId}", batchId, customerId);
+        var endpoint = $"/customers/{customerId}/queue/batches/{batchId}/assets";
+
+        var assets = await CallDlcsApiForJson(HttpMethod.Get, endpoint, null, cancellationToken);
+
+        return assets switch
+        {
+            null => [],
+            not null when assets.TryGetValue("member", out var member)
+                          && member.Type == JTokenType.Array => member.OfType<JObject>().ToList(),
+            _ => []
+        };
+    }
+
+    public async Task<IList<JObject>> GetCustomerImages(int customerId, ICollection<string> assetIds,
+        CancellationToken cancellationToken = default)
+    {
+        logger.LogTrace("Requesting images for customer {CustomerId}: {AssetIds}", customerId,
+            string.Join(",", assetIds));
+
+        if (assetIds.Count == 0)
+            return [];
+
+        var endpoint = $"/customers/{customerId}/allImages";
```

Comment:
There's an upper limit to the number of images that can be requested in 1 request from DLCS. It defaults to 500 but can differ https://github.com/dlcs/protagonist/blob/main/src/protagonist/API/Settings/ApiSettings.cs#L38. We should have a similar config value and chunk up into POSTs of max XXX image ids

Reply:
Cheers, I haven't noticed that!

Added analogous config value with same 500 default and turned the call into batched call. Again, could be at some point "improved", but that definitely gets in scope of "proper client".